src/core/stdc/stdlib.d

```diff
@@ -159,6 +159,17 @@ else
 ///
 void*   malloc(size_t size) pure;
 ///
+void*   pureMalloc(size_t size) pure
+{
+    const errno = pureGetErrno();
+    typeof(return) ret = malloc(size);
+    cast(void)pureSetErrno(errno);
+    return ret;
+}
+extern (C) pure pragma(mangle, "getErrno") int pureGetErrno(); // for internal use
+extern (C) pure pragma(mangle, "setErrno") int pureSetErrno(int); // for internal use
```

Comment:
Checking in with LDC developers @kinke @JohanEngelen, does this work on LDC?

BTW, I bumped into a related `extern (C)` issue on LDC - https://github.com/dlang/druntime/pull/1728#discussion_r96115755